File: src/main/java/architecture/goldenboughs_lib/init/LibSoundEvents.java
package architecture.goldenboughs_lib.init;

import architecture.goldenboughs_lib.core.GoldenBoughsLib;
import architecture.goldenboughs_lib.datagen.i18n.LibZhCn;
import net.minecraft.core.registries.BuiltInRegistries;
import net.minecraft.sounds.SoundEvent;
import net.neoforged.neoforge.registries.DeferredHolder;
import net.neoforged.neoforge.registries.DeferredRegister;

public final class LibSoundEvents {
	public static final DeferredRegister<SoundEvent> REGISTRY = GoldenBoughsLib.modRegister(BuiltInRegistries.SOUND_EVENT);

	private static DeferredHolder<SoundEvent, SoundEvent> registerForHolder(String id, String zhName, String location) {
		DeferredHolder<SoundEvent, SoundEvent> register = REGISTRY.register(id, () -> SoundEvent.createVariableRangeEvent(GoldenBoughsLib.modRl(location)));
		LibZhCn.addI18nSoundEventText(zhName, register);
		return register;
	}
}

File: src/main/kotlin/architecture/goldenboughs_lib/init/LibSoundEvents.kt
package architecture.goldenboughs_lib.init

import architecture.goldenboughs_lib.core.Lib.modRegister
import architecture.goldenboughs_lib.core.Lib.modRl
import architecture.goldenboughs_lib.datagen.i18n.LibZhCn
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.sounds.SoundEvent
import net.neoforged.neoforge.registries.DeferredHolder
import net.neoforged.neoforge.registries.DeferredRegister

object LibSoundEvents {

	@JvmField
	val REGISTRY: DeferredRegister<SoundEvent> = modRegister<SoundEvent>(BuiltInRegistries.SOUND_EVENT)

	private fun registerForHolder(
		id: String, zhName: String, location: String
	): DeferredHolder<SoundEvent, SoundEvent> {
		val register = REGISTRY.register(id) { -> SoundEvent.createVariableRangeEvent(modRl(location)) }
		LibZhCn.addI18nSoundEventText(zhName, register)
		return register
	}
}
